From: Gerald Schaefer <geraldsc@de.ibm.com>
Subject: topology: introduce arch specific SD_MC_INIT initializer
References: bnc#477666,LTC#51049

Symptom:     Up to 30% more cpu usage for some workloads.
Problem:     For some workloads the extra multicore scheduling domain causes
             additional cpu usage because of too optimistic assumptions when
             it is ok to migrate processes from one cpu to another. The default
             values for SD_MC_INIT don't work well on s390.
Solution:    Define an architecure specific SD_MC_INIT scheduling domain
             initializer which fixes the regression.

Acked-by: John Jolly <jjolly@suse.de>
---
 arch/s390/include/asm/topology.h |    2 ++
 1 file changed, 2 insertions(+)

Index: linux-sles11/arch/s390/include/asm/topology.h
===================================================================
--- linux-sles11.orig/arch/s390/include/asm/topology.h
+++ linux-sles11/arch/s390/include/asm/topology.h
@@ -28,6 +28,8 @@ static inline void s390_init_cpu_topolog
 };
 #endif
 
+#define SD_MC_INIT SD_CPU_INIT
+
 #include <asm-generic/topology.h>
 
 #endif /* _ASM_S390_TOPOLOGY_H */
